### ISSUE TITLE
[Question] New CLI prompts to install runtime twice if it is not installed, the first prompt now seems pointless

### ISSUE THREAD

**Issue by ahayzen** [flatpak/flatpak#1814]
## Linux distribution and version
Ubuntu 18.04 (using PPA)
## Flatpak version
0.99.1
## Description of the problem
When installing your first app, you get asked if you want to install the runtime, you say yes, then it asks again if you want to install the runtime, apps, extensions. This feels like a slightly strange UX.
## Steps to reproduce
1) Install flatpak on a new system or ensure you don't have the runtime for an app installed
2) Try to install an app (eg `org.gnome.Recipes`)
3) Notice you are asked if you want to install the runtime
4) Say yes
5) Now notice you are asked if you want to install the runtime, extensions and app with it's permissons.

This seems slightly odd and I am questioning if this could be merged into one question ? (basically removing the first as it seems potentially pointless now)

On a fresh Ubuntu 18.04 system this is what the user would have when installing their first app with the CLI.
```
$ flatpak install flathub org.gnome.Recipes
Required runtime for org.gnome.Recipes/x86_64/stable (runtime/org.gnome.Platform/x86_64/3.24) found in remote flathub
Do you want to install it? [y/n]: y
Installing in system:
org.gnome.Platform/x86_64/3.24             flathub 425f63d3d6b3
org.freedesktop.Platform.ffmpeg/x86_64/1.6 flathub b5f77229b85f
org.gtk.Gtk3theme.Ambiance/x86_64/3.22     flathub e34dff658694
org.gnome.Platform.Locale/x86_64/3.24      flathub a4b4636702f4
org.gnome.Recipes/x86_64/stable            flathub e6e666491d2b
  permissions: ipc, network, pulseaudio, wayland, x11
  file access: xdg-run/dconf, ~/.config/dconf:ro
  dbus access: ca.desrt.dconf
org.gnome.Recipes.Locale/x86_64/stable     flathub bdfb24090401
Is this ok [y/n]: y
...
```

**Comment by ahayzen** [comment]
After further thought, the first prompt is actually due to the runtime being potentially in multiple remotes. So the first question is asking you which remote should be chosen for the runtime.

I'm going to leave this issue option as a question if this UX could be improved, maybe in the case where the runtime is only in one remote it doesn't need to prompt ?

**Comment by alexlarsson** [comment]
Yeah, the first prompt is about verifying the source of the runtime. All flatpak installations are tied to a user-named source of apps/runtimes, and the user needs to explicitly give (or confirm) the source each time something is installed. That is part of the security model.

Could we combine this with the general confirmation in the case there is only one source? Possibly...

**Comment by aiamuzz** [comment]
Yeah even i have seen this on installing a new app, not always but yes i have come across it.

@alexlarsson ... If the prompt(choosing a runtime from a particular remote) is part of the security model ... then wouldn't it further simplify, if you recorded a user's preferences somewhere so that that remote becomes the default for downloading and installing runtimes ?

Its all about trusting the source ... so lets say I trust 'Flathub' as the source more than another one ... i can record/set THAT as my default ... that way the security 

Another factor for choosing a particular remote could also be the speed of downloads from a particular remote.

**Comment by alexlarsson** [comment]
I think we'll keep this, we need to know which runtime to use etc before we can do the full resolve that leads to the second promp.
